trace/doc.go

```diff
@@ -0,0 +1,82 @@
+// with a few fields using `beeline.AddField`. Occasionally add an additional
+// span to the trace that is automatically created and add fields to that.
+// However, as the beeline starts being used in more sophisticated applications,
+// or new wrapers are being written, more direct control is needed to manipulate
+// the spans and traces generated by the beeline.  Using the `trace` package
+// enables this more sophisticated use.
+//
+// The types of use that are enabled by using the trace package:
+//
+// 1) creating asynchronous spans; spans that outlive the main trace. These spans
+// are especially useful for goroutines that manage some background task after
+// the main user-facing chore has been completed. Examples of this are sending
+// an email or persisting accepted data.
+//
+// 2) creating traces with upstream parents when you are a downstream service.
+// The existing HTTP wrappers do this for you, but if your trace is getting
+// propagated via kafka or SQS or some other mechanism you may need to do this
+// yourself
+//
+// 3) adding fields that use a different naming scheme. Fields added via the
+// beeline are all namespaced under `app.`, which is convenient when you're
+// adding a few. When you have more complicated code to manage, it can be
+// useful to use your own naming scheme. Adding fields directly to the span or
+// trace objects allows you to specify the full field name with no prefix.
+//
+// Lifecycle
+//
+// A trace is made up of spans. A span represents a single unit of work (what
+// makes up a unit of work is up to the application). Spans come in two flavors,
+// synchronous (the default) and asynchronous. Synchronous spans finish before
+// their parents, async spans don't. Which you should use depends on the
+// structure of code - if the outer function will block until the inner function
+// returns, a sync span is appropriate. If the inner function is called in a
+// goroutine and expected to outlive the outer function, an async span fits
+// better.
+//
+// Spans should be created as children from an existing span. If there is no
+// current span, first create a new trace then get its root span and use that to
+// create subsequent spans. The beeline `StartSpan()` takes care of all of this
+// for you, but if you're using the trace package directly you need to manage
+// that bookkeeping.
+//
+// When should you create a new span? There are no strict rules, but there are a
+// few heuristics. If there is a process that will repeat in a loop (a batch or
+// something) and each run through the loop is important, make a span. If the
+// code being instrumented has enough attributes that are relevant to it
+// directly to warrant its own bag of data, make a new span. If all you're
+// interested in is a simple timer, it's often cleaner to add that timer to the
+// current existing span instead and save the overhead of a new span.
```

Comment:
"Save the overhead of a new span" makes it sound like creating spans is expensive, in a "will this slow down my super fast Go app?" sort of way.

As I understand it, the distinction is basically whether you want to see the sub-operation in a trace. Am I right in thinking that creating a span will also give you timer-like semantics on the parent span via the rollup support?

Reply:
removed the 'overhead of a new span part and just left it as unadorned advice.